viewing_party/party.py

```diff
@@ -0,0 +1,287 @@
+    # Grab the values at each title key and put into a set_1
+    for movie in user_data["watched"]:
+        user_set.add(movie["title"])
+
+    # this is a list - loop through this list of dictionaries
+    # in the dictionaries, access ["watched"], which is another list of dictionaries
+    # loop through the watched list to grab the ["title"] key and put that into a second set
+    for friend_data in user_data["friends"]:
+        for movie in friend_data["watched"]:
+            friend_set.add(movie["title"])
+    
+    # Compare both sets, getting the difference between set_1 and set_2
+    # Turn the resulting set into a list
+    # Return that list back to the function
+    unique_watched = list(user_set - friend_set)
+
+    new_list = []
+    # Create a dictionary for each title
+    # append that dictionary to unique watched with the key "title"
+    for movie in unique_watched:
+        new_list.append({"title": movie})
+
+    unique_watched = new_list
+    
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    """
+    input: A dictionary of user_data
+    output: A list of movies that a user's friends have watched, but the user has not
+    """
+
+    user_set = set()
+    friend_set = set()
+
+    # Collect all of the movie titles watched by original user_data
+    # Access with user_data[0] <= loop through this list of dictionaries
+    # Grab the values at each title key and put into a set_1
+    for movie in user_data["watched"]:
+        user_set.add(movie["title"])
+    # Then access the second element in user_data[1]
+    for friend_data in user_data["friends"]:
+        for movie in friend_data["watched"]:
+            friend_set.add(movie["title"])
+    # this is a list - loop through this list of dictionaries
+    # in the dictionaries, access ["watched"], which is another list of dictionaries
+    # loop through the watched list to grab the ["title"] key and put that into a second set
+    friends_unique_watched = list(friend_set - user_set)
+
+    new_list = []
+    # Create a dictionary for each title
+    # append that dictionary to unique watched with the key "title"
+    for movie in friends_unique_watched:
+        new_list.append({"title": movie})
+
+    friends_unique_watched = new_list
+    # Compare both sets, getting the difference between set_1 and set_2
+    # Turn the resulting set into a list
+    # Return that list back to the function
+
+    return friends_unique_watched
+
+def get_available_recs(user_data):
+    """
+    input: A dictionary of user_data
+    output: A list of movies that the user has not watched and that match the subscription platforms
+    the user uses
+    """
+    
+    recommended_movies = []
+    user_subscriptions = user_data["subscriptions"]
+
+    for friend in user_data["friends"]: # For each friend
+        for movie in friend["watched"]: # Loop through each movie the friend has watched
```

Comment:
As above, a helper function that returns `friends_set` may also be able to be used here.